src/lib/comms/include/sol-netctl.h

```diff
@@ -209,6 +298,60 @@ enum sol_netctl_state {
 };
 
 /**
+ * @brief agent input struct
+ *
+ * This struct contains the information of agent input.
+ */
+typedef struct sol_netctl_agent_input {
+    /**
+     * @brief The agent prompt type
+     */
+    enum sol_netctl_agent_prompt_type type;
+    /**
+     * @brief The agent input value
+     */
+    char *input;
+} sol_netctl_agent_input;
+
+/**
+ * @brief agent callback functions
+ *
+ * This struct contains the callback functions of agent.
+ */
+typedef struct sol_netctl_agent {
+    /**
+     * @brief connection error callback used to inform connection failure
+     *
+     * @param data the user data
+     * @param service the connection failure service
+     * @param error the error information
+     */
+    void (*report_error)(void *data, struct sol_netctl_service *service,
+        const char *error);
+    /**
+     * @brief connection input callback used to inform connection login input
+     *
+     * @param data the user data
+     * @param service the connection login input service
+     * @param vector the login input type
```

Comment:
rename from `vector` to `inputs` and say it should be a vector of `sol_net_agent_input`. Also see if you can use this as `sol_vector` instead of `ptr` vector (didn't check this part yet).